GOFevaluation/utils.py

```diff
@@ -316,13 +434,20 @@ def plot_equiprobable_histogram(data_sample, bin_edges, order=None,
                 ax.add_patch(rec)
                 j += 1
             i += 1
+
     fig = mpl.pyplot.gcf()
-    if nevents_expected is None:
-        label = 'Counts per Bin'
+    if(plot_mode == 'count_density'):
+        label = r'Counts per area in each bin'
+    elif(plot_mode == 'num_counts'):
+        label = r'Number of counts in each bin'
     else:
-        label = r'$\sigma$-deviation from expectation'
+        if nevents_expected is None:
+            label = 'Counts per Bin'
+        else:
+            label = r'$\sigma$-deviation from expectation'
```

Comment:
We need to clean this up as well. Also, maybe at the beginning we should enforce that `nevents_expectet` is not None of plot_mode is 'sigma_deviation'. Then we can remove this here (also it doesn't make sense otherwise)..